File: riskier/Services/LandUnitsService.cs
namespace riskier.Services;

public class LandUnitsService
{
  private readonly AccountsRepository _ar;

  public LandUnitsService(AccountsRepository ar)
  {
    _ar = ar;
  }

  internal void CreateInfantry(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 1;
    owner.TotalIndustry -= 1;
    owner.TotalAgriculture -= 3;
    owner.Infantry++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveInfantry(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 1;
    owner.TotalIndustry += 1;
    owner.TotalAgriculture += 3;
    if (owner.Infantry <= 0)
    {
      return;
    }
    owner.Infantry--;
    owner = _ar.Edit(owner);
  }

  internal void CreateSpecialForces(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 3;
    owner.TotalIndustry -= 1;
    owner.TotalAgriculture -= 3;
    owner.SpecialForces++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveSpecialForces(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 3;
    owner.TotalIndustry += 1;
    owner.TotalAgriculture += 3;
    if (owner.SpecialForces <= 0)
    {
      return;
    }
    owner.SpecialForces--;
    owner = _ar.Edit(owner);
  }

  internal void CreateMech(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 2;
    owner.TotalIndustry -= 2;
    owner.TotalAgriculture -= 3;
    owner.Mech++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveMech(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 2;
    owner.TotalIndustry += 2;
    owner.TotalAgriculture += 3;
    if (owner.Mech <= 0)
    {
      return;
    }
    owner.Mech--;
    owner = _ar.Edit(owner);
  }

  internal void CreateIfv(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 3;
    owner.TotalIndustry -= 3;
    owner.TotalAgriculture -= 1;
    owner.Ifv++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveIfv(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 3;
    owner.TotalIndustry += 3;
    owner.TotalAgriculture += 1;
    if (owner.Ifv <= 0)
    {
      return;
    }
    owner.Ifv--;
    owner = _ar.Edit(owner);
  }

  internal void CreateMbt(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 4;
    owner.TotalIndustry -= 4;
    owner.TotalAgriculture -= 1;
    owner.Mbt++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveMbt(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 4;
    owner.TotalIndustry += 4;
    owner.TotalAgriculture += 1;
    if (owner.Mbt <= 0)
    {
      return;
    }
    owner.Mbt--;
    owner = _ar.Edit(owner);
  }

  internal void CreateArtillery(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 1;
    owner.TotalIndustry -= 3;
    owner.TotalAgriculture -= 0;
    owner.Artillery++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveArtillery(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 1;
    owner.TotalIndustry += 3;
    owner.TotalAgriculture += 0;
    if (owner.Artillery <= 0)
    {
      return;
    }
    owner.Artillery--;
    owner = _ar.Edit(owner);
  }

  internal void CreateSsArtillery(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 2;
    owner.TotalIndustry -= 4;
    owner.TotalAgriculture -= 0;
    owner.SsArtillery++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveSsArtillery(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 2;
    owner.TotalIndustry += 4;
    owner.TotalAgriculture += 0;
    if (owner.SsArtillery <= 0)
    {
      return;
    }
    owner.SsArtillery--;
    owner = _ar.Edit(owner);
  }

  internal void CreateAntiAircraft(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 3;
    owner.TotalIndustry -= 1;
    owner.TotalAgriculture -= 0;
    owner.AntiAircraft++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveAntiAircraft(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 3;
    owner.TotalIndustry += 1;
    owner.TotalAgriculture += 0;
    if (owner.AntiAircraft <= 0)
    {
      return;
    }
    owner.AntiAircraft--;
    owner = _ar.Edit(owner);
  }

  internal void CreateSupplyTruck(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital -= 1;
    owner.TotalIndustry -= 0;
    owner.TotalAgriculture -= 1;
    owner.SupplyTruck++;
    owner = _ar.Edit(owner);
  }

  internal void RemoveSupplyTruck(string userId)
  {
    var owner = _ar.GetById(userId);
    owner.TotalCapital += 1;
    owner.TotalIndustry += 0;
    owner.TotalAgriculture += 1;
    if (owner.SupplyTruck <= 0)
    {
      return;
    }
    owner.SupplyTruck--;
    owner = _ar.Edit(owner);
  }
}

File: riskier.client/src/services/LandUnitsService.js
import { AppState } from "../AppState"
import { accountService } from "./AccountService";
import { api } from "./AxiosService"

class LandUnitsService {

  async addInfantry() {
    const res = await api.post('api/landUnits/infantry')
    accountService.getAccount()
  }

  async removeInfantry() {
    const res = await api.delete('api/landUnits/infantry')
    accountService.getAccount()
  }

  async addSpecialForces() {
    const res = await api.post('api/landUnits/specialForces')
    accountService.getAccount()
  }

  async removeSpecialForces() {
    const res = await api.delete('api/landUnits/specialForces')
    accountService.getAccount()
  }

  // Mech = Mechanized Infantry
  async addMech() {
    const res = await api.post('api/landunits/mech')
    accountService.getAccount()
  }

  async removeMech() {
    const res = await api.delete('api/landUnits/mech')
    accountService.getAccount()
  }

  // IFV = Infantry Fighting Vehicle
  async addIfv() {
    const res = await api.post('api/landunits/ifv')
    accountService.getAccount()
  }

  async removeIfv() {
    const res = await api.delete('api/landUnits/ifv')
    accountService.getAccount()
  }

  // MBT = Main Battle Tank
  async addMbt() {
    const res = await api.post('api/landunits/mbt')
    accountService.getAccount()
  }

  async removeMbt() {
    const res = await api.delete('api/landUnits/mbt')
    accountService.getAccount()
  }

  async addArtillery() {
    const res = await api.post('api/landunits/artillery')
    accountService.getAccount()
  }

  async removeArtillery() {
    const res = await api.delete('api/landUnits/artillery')
    accountService.getAccount()
  }

  // SsArtillery also known as missile artillery on board game
  async addSsArtillery() {
    const res = await api.post('api/landunits/ssArtillery')
    accountService.getAccount()
  }

  async removeSsArtillery() {
    const res = await api.delete('api/landUnits/ssArtillery')
    accountService.getAccount()
  }

  async addAntiAircraft() {
    const res = await api.post('api/landunits/antiAircraft')
    accountService.getAccount()
  }

  async removeAntiAircraft() {
    const res = await api.delete('api/landUnits/antiAircraft')
    accountService.getAccount()
  }

  async addSupplyTruck() {
    const res = await api.post('api/landunits/supplyTruck')
    accountService.getAccount()
  }

  async removeSupplyTruck() {
    const res = await api.delete('api/landUnits/supplyTruck')
    accountService.getAccount()
  }
}
export const landUnitsService = new LandUnitsService()